pkg/controllers/dynakube/istio/reconciler.go

```diff
@@ -62,35 +67,89 @@ func (r *reconciler) ReconcileCodeModuleCommunicationHosts(ctx context.Context,
 
 	oneAgentCommunicationHosts := oaconnectioninfo.GetCommunicationHosts(dynakube)
 
+	const conditionComponent = "CodeModule"
+
 	err := r.reconcileCommunicationHostsForComponent(ctx, oneAgentCommunicationHosts, OneAgentComponent)
 	if err != nil {
+		setServiceEntryFailedConditionForComponent(dynakube.Conditions(), conditionComponent, err)
+
 		return err
 	}
 
+	if len(oneAgentCommunicationHosts) > 0 {
+		setServiceEntryUpdatedConditionForComponent(dynakube.Conditions(), conditionComponent)
+	} else {
+		meta.RemoveStatusCondition(dynakube.Conditions(), getConditionTypeName(conditionComponent))
+	}
+
 	return nil
 }
 
 func (r *reconciler) ReconcileActiveGateCommunicationHosts(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) error {
+	const conditionComponent = "ActiveGate"
+
 	log.Info("reconciling istio components for activegate communication hosts")
 
 	if dynakube == nil {
-		return errors.New("can't reconcile oneagent communication hosts of nil dynakube")
+		return errors.New("can't reconcile activegate communication hosts of nil dynakube")
 	}
 
 	if !dynakube.NeedsActiveGate() {
+		if isIstioConfigured(dynakube, conditionComponent) {
+			log.Info("ActiveGate disabled, cleaning up")
+
+			return r.CleanupIstio(ctx, dynakube, conditionComponent)
+		} else {
+			return nil
+		}
+	}
+
+	if !conditions.IsOutdated(r.timeProvider, dynakube, getConditionTypeName(conditionComponent)) {
```

Comment:
I like this, could we do this for the ` ReconcileCodeModuleCommunicationHosts`?

you should just need to change the `if` to `!dk.NeedAppInjection()`